File: leetcode/p821.cpp
class Solution {
public:
    vector<int> shortestToChar(string s, char c) {
        vector<int> res;
        res.assign(s.size(), INT_MAX);
        int distance = INT_MAX;
        for (int i=0;i<s.size();i++) {
            if (s[i] == c) {
                distance = 0;
            }
            if (res[i] > distance) {
                res[i] = distance;
            }
            if (distance < INT_MAX) {
                distance++;
            }
        }
        for (int i=s.size()-1;i>=0;i--) {
            if (s[i] == c) {
                distance = 0;
            }
            if (res[i] > distance) {
                res[i] = distance;
            }
            if (distance < INT_MAX) {
                distance++;
            }
        }
        return res;
    }
};

File: leetcode/p821.py
class Solution:
    def bfs(self, i):
        if i>0:
            if self.tab[i-1]>self.tab[i]+1:
                self.tab[i-1] = self.tab[i]+1
                self.queue.append(i-1)
        if i<len(self.tab)-1:
            if self.tab[i+1]>self.tab[i]+1:
                self.tab[i+1] = self.tab[i]+1
                self.queue.append(i+1)
    
    def shortestToChar(self, s: str, c: str) -> List[int]:
        self.tab = [float('inf')]*len(s)
        self.queue = [i for i in range(len(s)) if s[i]==c]
        for i in self.queue:
            self.tab[i] = 0
        while self.queue:
            i = self.queue.pop(0)
            self.bfs(i)
        return self.tab
